### PR TITLE
Experiments/models

### PR DESCRIPTION
**Pull request by oliviamartiin**
This script implements and evaluates classical machine learning models for Language Identification (LID). It follows the provided task description by training, evaluating, and comparing different models on a dataset. Below is a breakdown of what the code does and how it aligns with the given task:

Key Functions and Their Purpose

1. load_and_prepare_data()

Loads the dataset (wili_subset_khan.csv).
Preprocesses the data using preprocess().
Splits it into training, validation, and test sets.
Ensures proper working directory settings.
🛠 Task Alignment: Prepares data for training and evaluation.

2. run_experiment()

This function is the core of the evaluation process. For a given machine learning model, it:
- Vectorizes the dataset using vectorize_data(), based on:
1. Character-level n-grams
2. Word-level n-grams
3. Different vocabulary sizes (vocab).
- Trains the model using the training set.
- Predicts the labels for the test set.
- Evaluates performance using:
1. classification_report(): Precision, recall, F1-score.
2. confusion_matrix(): Identifies misclassification trends.
- Visualizes decision boundaries using PCA:
- Reduces features to 2D for plotting.
- Colors indicate language labels.
- Performs error analysis:
1. Displays misclassified examples.
🛠 Task Alignment:
- Implements multiple feature extraction methods.
- Evaluates models using classification reports and confusion matrices.
- Provides PCA-based visualizations of decision boundaries.
- Conducts error analysis.

3. run_naive_bayes()
- Trains Multinomial Naïve Bayes with different:
1. N-gram types (word, char).
2. Vocabulary sizes (256, 512, 1024).
3. Calls run_experiment() for each setting.
🛠 Task Alignment: Implements and tests MultinomialNB, fulfilling the requirement for a Naïve Bayes baseline.

4. run_svm()
- Trains an SVM classifier with:
1. Character-level and word-level n-grams.
2. Different vocabulary sizes.
3. Uses SVC(probability=True).
🛠 Task Alignment: Implements SVM for comparison.

5. run_logistic_regression()
- Trains Logistic Regression using:
1. Different n-gram types.
2. Various vocabulary sizes.
3. Sets max_iter=2000 for stability.
🛠 Task Alignment: Implements Logistic Regression, another required baseline.

### CODE REVIEW

**Comment by devWhyqueue** [comment]
Code to run these experiments already exists in the project. You just have to run the CLI with different CLI options. To change the model you just have to modify one line:

```python
model = MultinomialNB()
```

I would argue that the only reasonable code changes would be one of the following:

1. **Just one line:** Insert the best performing model into the line mentioned above, change nothing else.
2. **Add CLI option:** Add CLI option "model" to `baseline` command which can take values `[naive_bayes, svm, log_regression]`

**Review by devWhyqueue — CHANGES_REQUESTED** [review]
See comment above
